product_unique_serial/wizard/stock_serial.py

```diff
@@ -79,31 +80,33 @@ def default_get(self, field_list):
 
     @api.onchange('serial_ids')
     def onchange_serial(self):
-        serial = []
-
-        for serial_name in self.serial_ids:
-            if serial_name.serial in serial:
-                return {
-                    'warning': {
-                        'title': _('Warning'),
-                        'message': _(
-                            'The Serial number {} already captured'.format(
-                                serial_name.serial.encode('utf-8')))
-                    }}
-            else:
-                serial.append(serial_name.serial)
+        serials = [serial.serial for serial in self.serial_ids]
```

Comment:
What about create a `unique-sql-constraint` for `stock.serial.line` to avoid duplicated?
You could have the database healthy if you are using a different way to insert records (import csv files, create from webservices, sql inserts, create from other models)

I mean `UNIQUE('serial_id', 'serial')` from `stock.serial.line`
You can see real example here: https://github.com/odoo/odoo/blob/bad26aae2e175868bc082c11ddcd1131946414ce/addons/stock/models/stock_pack_operation.py#L267-L270